test/Libraries/PackageManagerTests/PackageUploadBuilderTests.cs

```diff
@@ -29,6 +32,68 @@ private IPackageUploadBuilder BigPackageUploadBuilderMock()
             return new PackageUploadBuilder(pdb.Object, zipper.Object);
         }
 
+        private class directoryTestClass : IDirectoryInfo
+        {
+            public string FullName { get; set; }
+
+        }
+
+        #endregion
+
+        #region GregFileUtilsTests
+            [Test]
+            public void FileCompressorZipsSamplePackageDirectoryToValidArchive()
+        {
+            var executingLocation = new DirectoryInfo(Assembly.GetExecutingAssembly().Location);
+            var testDirectory = Path.Combine(executingLocation.Parent.Parent.Parent.Parent.FullName, "test");
+            var aPackagePath =  new DirectoryInfo(Path.Combine(testDirectory, "pkgs","sampleExtension"));
+            var preZipfileCount = aPackagePath.GetFiles().Length;
+            var preZipDiskSize = aPackagePath.GetFiles().Sum(x => x.Length);
+
+            var packageDir = new directoryTestClass()
+            {
+                FullName = aPackagePath.FullName
+            };
+
+            var compressor = new MutatingFileCompressor();
+            var zipPath = compressor.Zip(packageDir);
+            Assert.AreEqual(2758, zipPath.Length);
+
+            //unzip the zipped directory
+            var unzipPath = Greg.Utility.FileUtilities.UnZip(zipPath.Name);
+
+            Assert.AreEqual(preZipDiskSize, new DirectoryInfo(unzipPath).GetFiles().Sum(x => x.Length));
+            Assert.AreEqual(preZipfileCount, new DirectoryInfo(unzipPath).GetFiles().Length);
```

Comment:
Not super familiar with what happens behind the scenes when zipping/unzipping, but is it at all likely that the size and number of files could be consistent yet still have been corrupted somehow? Would it be insane to check individual bytes?

Reply:
I think in general if we get this far in the test we're good, previously this bug would have thrown an exception.

Reply:
Late to the party here, but I think it would be reasonable to have a test that assembles a folder, zips the folder, unzips the zip into another folder, and compares the contents to ensure they're identical.

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/how-to-compare-the-contents-of-two-folders-linq

Reply:
@Dewb ended up doing that in another PR using md5 hashes for each file.
https://github.com/DynamoDS/Dynamo/pull/9281